tests/tape/tapes/test_tape.py

```diff
@@ -498,6 +498,24 @@ def test_setting_free_parameters(self, make_tape):
             params[4],
         ]
 
+    def test_setting_parameters_unordered(self, make_tape, monkeypatch):
+        """Test that an 'unordered' trainable_params set does not affect
+        the setting of parameter values"""
+        tape, params = make_tape
+        new_params = [-0.654, 0.3]
+
+        with monkeypatch.context() as m:
+            m.setattr(tape, "_trainable_params", {3, 1})
+            tape.set_parameters(new_params)
+
+        assert tape.get_parameters(trainable_only=False) == [
```

Comment:
How about moving the `assert` into the `monkeypatch` context and querying `tape.get_parameters(trainable_only=True)`? Think it would make the test case a bit clearer.

Reply:
done!